edgelet/iotedged/src/app.rs

```diff
@@ -75,13 +80,30 @@ fn init_common(running_as_windows_service: bool) -> Result<Settings, Error> {
     Ok(settings)
 }
 
-pub fn init() -> Result<Settings, Error> {
-    init_common(false)
+#[cfg(feature = "runtime-docker")]
+fn start_message() {
+    info!("Starting Azure IoT Edge Security Daemon");
+    info!("Version - {}", edgelet_core::version_with_source_version());
+}
+#[cfg(feature = "runtime-kubernetes")]
+fn start_message() {
+    info!("Starting Azure IoT Edge Security Daemon - Kubernetes mode");
+    info!("Version - {}", edgelet_core::version_with_source_version());
+}
+
+pub fn init<M>() -> Result<Settings, Error>
```

Comment:
What's the point of taking `M` ? You're not doing any type-level specialization with it. The actual condition (choice of start message) is done via `cfg`.

Reply:
It's an artifact from a different iteration (I was originally returning Result<M::Settings, Error>).  I can remove it.